contracts/seed/Seed.sol

```diff
@@ -176,58 +177,63 @@ contract Seed {
         minimumReached    = false;
         maximumReached    = false;
 
-        // Same calculation as we did at SeedFactory.deploySeed()
-        uint256 seedAmountSent = (_softHardThresholds[1].div(_price)).mul(10**18);
-        seedAmountSent = seedAmountSent.add((seedAmountSent.mul(uint256(PPM))).mul(_fee).div(PPM100));
-
-        seedRemainder     = seedAmountSent;
-        seedAmountAtStart = seedAmountSent;
+        seedAmountRequired = (hardCap.div(_price)).mul(10**18);
+        seedForFeeRequired = seedAmountRequired.mul(_fee).div(100);
+        seedRemainder     = seedAmountRequired;
+        feeSeedRemainder  = seedForFeeRequired;
     }
 
     /**
-      * @dev                     Buy and lock seed tokens.
-      * @param _fundingAmount       The amount of seed tokens to buy.
+      * @dev                     Buy seed tokens.
+      * @param _fundingAmount    The amount of funding tokens to contribute.
     */
     function buy(uint256 _fundingAmount) public isActive allowedToBuy returns(uint256, uint256) {
+        if(!isFunded) {
+            require(seedToken.balanceOf(address(this)) >= seedAmountRequired.add(seedForFeeRequired),
+                "Seed: sufficient seeds not provided");
+            isFunded = true;
+        }
         // fundingAmount is an amount of fundingTokens required to buy _seedAmount of SeedTokens
         uint256 seedAmount = (_fundingAmount.mul(PCT_BASE)).div(price);
 
         // Funding Token balance of this contract;
         uint256 fundingBalance = fundingCollected;
 
         // feeAmount is an amount of fee we are going to get in seedTokens
-        uint256 feeAmount = (seedAmount.mul(uint256(PPM))).mul(fee).div(PPM100);
```

Comment:
PPM and PPM100 constants should be removed too

Reply:
Done